src/renderer/components/output.tsx

```diff
@@ -30,121 +33,195 @@ interface CommandsProps {
+   * mount, not React's.
+   */
+  public async editorDidMount(editor: MonacoType.editor.IStandaloneCodeEditor) {
+    const { editorDidMount } = this.props;
+
+    await this.setContent(this.props.appState.output);
+
+    // Notify other editors that the initial editor was mounted
+    if (editorDidMount) {
+      editorDidMount(editor);
     }
   }
 
   /**
-   * Render the timestamp
-   *
-   * @param {number} ts
-   * @returns {string}
+   *  Set Monaco Editor's value.
    */
-  public renderTimestamp(ts: number): string {
-    const { renderTimestamp } = this.props;
+  public async UNSAFE_componentWillReceiveProps(newProps: CommandsProps) {
+    await this.setContent(newProps.appState.output);
+  }
 
-    if (renderTimestamp) {
-      return renderTimestamp(ts);
-    } else {
-      return new Date(ts).toLocaleTimeString();
+  /**
+   * Initialize Monaco.
+   */
+  public async initMonaco() {
+    const { monaco, monacoOptions: monacoOptions } = this.props;
+    const ref = this.outputRef.current;
+    if (ref) {
+      this.setupMonacoLanguage(monaco);
+      this.editor = monaco.editor.create(ref, {
+        language: this.language,
+        theme: 'main',
+        readOnly: true,
+        contextmenu: false,
+        automaticLayout: true,
+        model: null,
+        ...monacoOptions,
+      });
+
+      await this.editorDidMount(this.editor);
     }
   }
 
   /**
+   * Destroy Monaco.
+   */
+  public destroyMonaco() {
+    if (typeof this.editor !== 'undefined') {
+      console.log('Editor: Disposing');
+      this.editor.dispose();
+    }
+  }
+
+  public toggleConsole() {
+    /**
+     * Type guard to check whether a react-mosaic node is a parent in the tree
+     * or a leaf. Leaf nodes are represented by the string value of their ID,
+     * whereas parent nodes are objects containing information about the nested
+     * binary tree.
+     * @param node A react-mosaic node
+     * @returns Whether that node is a MosaicParent or not
+     */
+    const isParentNode = (
+      node: MosaicNode<WrapperEditorId> | null,
+    ): node is MosaicParent<WrapperEditorId> => {
+      return (node as MosaicParent<WrapperEditorId>)?.direction !== undefined;
+    };
+
+    const { isConsoleShowing } = this.props.appState;
+
+    if (this.context.mosaicActions) {
+      const mosaicTree = this.context.mosaicActions.getRoot();
+      if (isParentNode(mosaicTree)) {
+        // splitPercentage defines the percentage of space the first panel takes
+        // e.g. 25 would mean the two children panels are split 25%/75%
+        if (!isConsoleShowing) {
+          mosaicTree.splitPercentage = 0;
+        } else if (mosaicTree.splitPercentage === 0) {
+          mosaicTree.splitPercentage = 25;
+        }
+        this.context.mosaicActions.replaceWith([], mosaicTree);
+      }
+    }
+  }
+
+  /**
+   * Set up Monaco Language.
+   */
+  private setupMonacoLanguage(monaco: typeof MonacoType) {
```

Comment:
This could be more descriptive about what language we're setting up, e.g. explaining that we're tokenizing the date.

Hmm, maybe something like `setupDataDelimitedLanguage()`? That feels a little clumsy so don't feel compelled to use that verbatim, but something more descriptive than 'setupMonacoLanguage`

Reply:
setupCustomOutputEditorLanguage? :P